janusgraph-core/src/main/java/org/janusgraph/graphdb/database/StandardJanusGraph.java

```diff
@@ -735,17 +787,24 @@ public ModificationSummary prepareCommit(final Collection<InternalRelation> adde
             StaticBuffer vertexKey = idManager.getKey(vertexId);
             mutator.mutateEdges(vertexKey, additions, deletions);
         }
+    }
 
-        //6) Add index updates
+    /**
+     * Add index updates
+     *
+     * @return `true` if there was any mixed index update
+     */
+    private boolean prepareCommitIndexUpdatesAndCheckIfAnyMixedIndexUsed(final List<IndexUpdate> indexUpdates,
+                                                                         final BackendTransaction mutator) throws BackendException {
         boolean has2iMods = false;
         for (IndexUpdate indexUpdate : indexUpdates) {
             assert indexUpdate.isAddition() || indexUpdate.isDeletion();
             if (indexUpdate.isCompositeIndex()) {
                 final IndexUpdate<StaticBuffer,Entry> update = indexUpdate;
                 if (update.isAddition())
-                    mutator.mutateIndex(update.getKey(), Lists.newArrayList(update.getEntry()), KCVSCache.NO_DELETIONS);
+                    mutator.mutateIndex(update.getKey(), Collections.singletonList(update.getEntry()), KCVSCache.NO_DELETIONS);
                 else
-                    mutator.mutateIndex(update.getKey(), KeyColumnValueStore.NO_ADDITIONS, Lists.newArrayList(update.getEntry()));
+                    mutator.mutateIndex(update.getKey(), KeyColumnValueStore.NO_ADDITIONS, Collections.singletonList(update.getEntry()));
```

Comment:
`Lists.newArrayList` is not equivalent to `Collections.singletonList` but `mutateIndex` method copies the list internally. Thus, unmodifiable list is good here and on the line above.